### PR TITLE
Feature/species list tool

### PR DESCRIPTION
**Pull request by jmalczyk**
Adds EOL buttons that link to species pages on EOL to the species list tool, and wiki buttons just to be consistent.

### CODE REVIEW

File: app/js/mol.map.query.js [hunk 1]
```diff
@@ -305,6 +312,26 @@ mol.modules.map.query = function(mol) {
...
                          _.each(
+                             $('.eol',$(infoWindow.content)),
+                             function(button) {
+                                 if(button.value==''||button.value=='null') {
+                                    $(button).click(
+                                         function(event) {
+                                            var win = window.open('http://eol.org/search/?q={0}'.format($(this).data('sciname')));
+                                            win.focus();
+                                        }
+                                    )
```
Comment (eightysteele) [inline app/js/mol.map.query.js]: Add semicolon.

File: app/js/mol.map.query.js [hunk 2]
```diff
@@ -430,6 +457,7 @@ mol.modules.map.query = function(mol) {
                         '           <button class="ecoregion" value=" AND p.type=\'ecoregion\' "><img title="Click to use Regional checklists for query." src="/static/maps/search/ecoregion.png"></button>' +
                         '       </span>'+
                         '   </div>' +
+                        //'   <div>> Species List Tool</div>' +
```
Comment (eightysteele) [inline app/js/mol.map.query.js]: Why is this commented out? If we don't need it, let's remove it.

**Comment by eightysteele** [comment]
There are some formatting problems in `mol.map.query.js`. Lines 100, 133, 213, 232, and more are missing semicolons. The `stats` variable on line 236 is undeclared which makes it global (so just need to declare it as a var at top of function). Please double check all indents as well. They should be 4 spaces (no tabs). I'm seeing a lot of extra indentations.

File: app/js/mol.map.query.js [hunk 3]
```diff
@@ -305,6 +312,26 @@ mol.modules.map.query = function(mol) {
                              }
                          );
                          _.each(
+                             $('.eol',$(infoWindow.content)),
+                             function(button) {
+                                 if(button.value==''||button.value=='null') {
+                                    $(button).click(
+                                         function(event) {
+                                            var win = window.open('http://eol.org/search/?q={0}'.format($(this).data('sciname')));
```
Comment (eightysteele) [inline app/js/mol.map.query.js]: Is this cross browser? Maybe use JQuery to open a window instead.

File: app/js/mol.map.query.js [hunk 4]
```diff
@@ -305,6 +312,26 @@ mol.modules.map.query = function(mol) {
...
+                                            win.focus();
+                                        }
+                                    )
+                                 } else {
+                                    $(button).click(
+                                         function(event) {
+                                            var win = window.open('http://eol.org/pages/{0}/overview'.format(this.value));
```
Comment (eightysteele) [inline app/js/mol.map.query.js]: Same comment about cross browser. Double check if there's a JQuery way to do this.

File: app/js/mol.map.query.js (hunk 3 above)
Comment (jmalczyk) [inline app/js/mol.map.query.js]: As far as I can tell, there is no JQuery method for opening a window, and window.open is supported across all browsers.

Comment (eightysteele) [inline app/js/mol.map.query.js]: Good.
On Jul 19, 2012 10:10 AM, "Jeremy Malczyk" <
reply@reply.github.com>
wrote:

File: app/js/mol.map.query.js (hunk 2 above)
Comment (jmalczyk) [inline app/js/mol.map.query.js]: Was a place holder for another feature, removed.

File: app/js/mol.map.query.js (hunk 1 above)
Comment (jmalczyk) [inline app/js/mol.map.query.js]: Semicolon, added. Here and elsewhere.

**Comment by eightysteele** [comment]
@jmalczyk looking better dude. I fixed a bunch of additional formatting stuff. Also added some TODOs. Big one is the if statement on line 210 which is a complete beast. That thing needs to be broken down a bit into a few different functions. Right now it's too hard to reason about. At the least, add some docs in there and sub an issue to fix that up.

**Comment by jmalczyk** [comment]
Added issue #114

**Pull request merged**